src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -40,6 +40,12 @@ protected SCMNavigatorDescriptor(Class<? extends SCMNavigator> clazz) {
         super(clazz);
     }
 
+    public abstract String getDescription();
+
+    public abstract String getCategoryId();
+
+    public abstract String getIconFilePathPattern();
```

Comment:
You may not add `abstract` methods to an interface like this. It would not be compatible for existing subtypes.